### ISSUE TITLE
Support for Adaboost Regressor

### ISSUE THREAD

**Issue by M-Schreurs** [shap/shap#2727]
Similar to the issue solved in #335, I was wondering if there is some way to implement Adaboost Regressor in Shap values.
I tried the fix for the Classifier, but the force plot and summary don't make a lot of sense (all my SHAP values are tiny, < 10^-16).

**Comment by M-Schreurs** [comment]
I managed to get it to work myself by looking at the other regressors and the solution in #335 ([posted here on StackExchange](https://stackoverflow.com/questions/60433389/how-to-calculate-shap-values-for-adaboost-model)).

`Option 1: scaling = 1.0 . len(model.estimators_)`

        elif safe_isinstance(model, ("sklearn.ensemble.AdaBoostClassifier", "sklearn.ensemble._weighted_boosting.AdaBoostClassifier", "imblearn.ensemble.RUSBoostClassifier", "imblearn.ensemble._weight_boosting.RUSBoostClassifier")):
            assert hasattr(model, "estimators_"), "Model has no `estimators_`! Have you called `model.fit`?"
            self.internal_dtype = model.estimators_[0].tree_.value.dtype.type
            self.input_dtype = np.float32
            scaling = 1.0 / len(model.estimators_) # output is average of trees
            self.trees = [SingleTree(e.tree_, normalize=True, scaling=scaling, data=data, data_missing=data_missing) for e in model.estimators_]
            self.objective = objective_name_map.get(model.base_estimator_.criterion, None) #This line is done to get the decision criteria, for example gini.
            self.tree_output = "probability"
        elif safe_isinstance(model, ("sklearn.ensemble._weighted_boosting.AdaBoostRegressor", "sklearn.ensemble.AdaBoostRegressor")):
            assert hasattr(model, "estimators_"), "Model has no `estimators_`! Have you called `model.fit`?"
            self.internal_dtype = model.estimators_[0].tree_.value.dtype.type
            self.input_dtype = np.float32
            scaling = 1.0 / len(model.estimators_) # output is average of trees
            self.trees = [SingleTree(e.tree_, scaling=scaling, data=data, data_missing=data_missing) for e in model.estimators_]
            self.objective = objective_name_map.get(model.base_estimator_.criterion, None) #This line is done to get the decision criteria, for example gini.
            self.tree_output = "raw_value"


` Option 2: scaling = model.estimator_weights_/sum(model.estimator_weights_) `


        elif safe_isinstance(model, ("sklearn.ensemble.AdaBoostClassifier", "sklearn.ensemble._weighted_boosting.AdaBoostClassifier", "imblearn.ensemble.RUSBoostClassifier", "imblearn.ensemble._weight_boosting.RUSBoostClassifier")):
            assert hasattr(model, "estimators_"), "Model has no `estimators_`! Have you called `model.fit`?"
            self.internal_dtype = model.estimators_[0].tree_.value.dtype.type
            self.input_dtype = np.float32
            self.trees = [SingleTree(e.tree_, normalize=True, scaling=weight, data=data, data_missing=data_missing) for e, weight in zip(model.estimators_, model.estimator_weights_/sum(model.estimator_weights_))]
            self.objective = objective_name_map.get(model.base_estimator_.criterion, None) #This line is done to get the decision criteria, for example gini.
            self.tree_output = "probability"
        elif safe_isinstance(model, ("sklearn.ensemble._weighted_boosting.AdaBoostRegressor", "sklearn.ensemble.AdaBoostRegressor")):
            assert hasattr(model, "estimators_"), "Model has no `estimators_`! Have you called `model.fit`?"
            self.internal_dtype = model.estimators_[0].tree_.value.dtype.type
            self.input_dtype = np.float32
            self.trees = [SingleTree(e.tree_, scaling=weight, data=data, data_missing=data_missing) for e, weight in zip(model.estimators_, model.estimator_weights_/sum(model.estimator_weights_))]
            self.objective = objective_name_map.get(model.base_estimator_.criterion, None) #This line is done to get the decision criteria, for example gini.
            self.tree_output = "raw_value"

**Comment by NikolaySimankov** [comment]
@M-Schreurs, looks like it could solve my issue, can you please Fork this update? 
Thx

**Comment by M-Schreurs** [comment]
Hi @NikolaySimankov,
you can find a Fork with the edits here:
https://github.com/M-Schreurs/shap

**Comment by NikolaySimankov** [comment]
Hi @M-Schreurs, 
Nice work, thank you !
